alpa/device_mesh.py

```diff
@@ -715,11 +716,18 @@ def sync(self):
     def check_alive():
         return True
 
-    def shutdown(self):
+    def shutdown(self, forced=False):
+        if not self.launched:
+            return
         self.sync()
         self.buffers.clear()
         self.executables.clear()
         self.distributed_client.shutdown()
+        # shutdown DaemonMoveWorker
+        if not forced:
+            ray.get(self.move_worker.shutdown.remote())
```

Comment:
Some comments here:
1. Rename `move_worker.shutdown` -> `move_worker.sync`. Because it's not shutting down the move_worker.
2. Instead of only syncing the save worker when shutdown, add an explicit `sync_move_worker` call to synchronize the sync worker and call `sync_move_worker` here.
3. Eventually, we can expose an API in `PhysicalDeviceMesh` that lets a user explicitly synchronize all the model saving workers. This might be useful in the missing unit test below.